packages/augur-sdk/src/state/db/MarketDB.ts

```diff
@@ -85,37 +86,43 @@ export class MarketDB extends DerivedDB {
     }
   }
 
-  syncOrderBooks = async (syncing: boolean): Promise<void> => {
+  syncOrderBooks = async (): Promise<void> => {
     const highestSyncedBlockNumber = await this.syncStatus.getHighestSyncBlock(this.dbName);
     const documents = [];
 
-    const currentOrderLogs = await this.stateDB.CurrentOrders.where("blockNumber").aboveOrEqual(highestSyncedBlockNumber).toArray();
+    const currentZeroXOrders = await this.stateDB.ZeroXOrders.toArray();
 
-    if (currentOrderLogs.length < 1) return;
+    if (currentZeroXOrders.length < 1) return;
 
-    const marketIds: string[] = _.uniq(_.map(currentOrderLogs, 'market')) as string[];
-    const highestBlockNumber: number = _.max(_.map(currentOrderLogs, 'blockNumber')) as number;
+    const marketIds: string[] = _.uniq(_.map(currentZeroXOrders, 'market')) as string[];
     const marketsData = await this.stateDB.Markets.where("market").anyOf(marketIds).toArray();
-
-    const marketDataById = _.keyBy(marketsData, 'market');
-
     const reportingFeeDivisor = await this.augur.contracts.universe.getReportingFeeDivisor_();
     // TODO Get ETH -> DAI price via uniswap when we integrate that as an oracle
     const ETHInAttoDAI = new BigNumber(200).multipliedBy(10**18);
 
+    const marketDataById = _.keyBy(marketsData, 'market');
+
+    const now = Math.floor(Date.now() / 1000);
+
+    // only check market if it hasn't already been checked in the last 30s
```

Comment:
This check is good for the debouncing sort of behavior but will still update more markets than we need to check.

I think we can address this by actually looking at the order events that are provided when handling the subscription event here: `this.augur.events.subscribe('DerivedDB:updated:ZeroXOrders',` and get the unique market ids out of that to see what needs updating